docs/articles/documentation/test-api/selecting-page-elements/selectors.md

```diff
@@ -0,0 +1,313 @@
+
+## Using Selectors to Define Action Targets
+
+You can pass selectors to [test actions](../actions/index.md) to use the returned element as the action target.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page('http://www.example.com/');
+
+const getLastItem = Selector(() => document.querySelector('.toc-item:last-child'));
+
+test('My Test', async t => {
+    await t.click(getLastItem);
+});
+```
+
+In this instance, the selector will be called with no arguments.
+
+You can also pass a promise returned by a selector if you need to call it with arguments.
+
+```js
+import { Selector } from 'testcafe';
+
+const getElementById = Selector(id => document.getElementById(id));
+
+fixture `My fixture`
+    .page('http://www.example.com/');
+
+test('My Test', async t => {
+    await t.click(getElementById('submit-button'));
+});
+```
+
+DOM element snapshots can also be passed to test actions.
+
+```js
+import { Selector } from 'testcafe';
+
+const getElementById = Selector(id => document.getElementById(id));
+
+fixture `My fixture`
+    .page('http://www.example.com/');
+
+test('My Test', async t => {
+    const topMenuSnapshot = await getElementById('top-menu');
+
+    await t.click(topMenuSnapshot);
+});
+```
+
+In this instance, the selector that was used to fetch this snapshot will be called once again.
+
+When a selector, its promise or DOM element snapshot is passed to an action, TestCafe waits for the target element to appear
+in the DOM and become visible before this action is executed. If this does not happen
+within the selector timeout, which is specified in test code by using the [timeout](selector-options.md#optionstimeout) option
+or when launching tests via [API](../../using-testcafe/programming-interface/runner.md#run)
+or [CLI](../../using-testcafe/command-line-interface.md#--selector-timeout-ms), the test fails.
+
+## Calling Selectors from Node.js Callbacks
+
+Selectors need access to the [test controller](../test-code-structure.md#test-controller) to be executed.
+When called right from the test function, they implicitly obtain the test controller.
+
+However, if you need to call a selector from a Node.js callback that fires during the test run,
+you will have to manually bind it to the test controller.
+
+Use the [boundTestRun](selector-options.md#optionsboundtestrun) option for this.
+
+```js
+import { fs } from 'fs';
+import { expect } from 'chai';
+import { ClientFunction } from 'testcafe';
+
+fixture `My fixture`
+    .page('http://www.example.com/');
+
+const getDataFromClient = ClientFunction(() => getSomeData());
+
+test('Check client data', async t => {
+    const boundFunc = getDataFromClient.with({ boundTestRun: t });
+
+    const equal = await new Promise(resolve => {
+        fs.readFile('/home/user/tests/reference/clientData.json', (err, data) => {
+            resolve(JSON.stringify(await boundFunc()) === data);
```

Comment:
E.g. you have online document viewer. You refresh document from server and would like to see if image that you added to document appeared in the viewer